### PR TITLE
Upgrade Kubebuilder Version to v3

### PR DESCRIPTION
**Pull request by J12934**
This PR upgrades the kubebuilder version used by the secureCodeBox Operator to v3.
This also includes the migration of the CRD's from `apiextensions.k8s.io/v1beta1` to `apiextensions.k8s.io/v1`

## Upgrade notes

- The CRD's are now using `apiextensions.k8s.io/v1` which requries at least Kubernetes Version 1.16 or higher.
- The Operator now uses the new kubebuilder v3 command line flag for enabling leader election and setting the metrics port. If you are using the official secureCodeBox Helm Charts for your Deployment this has been updated automatically. If you are using a custom deployment you have to change the `--enable-leader-election` flag to `--leader-elect` and `--metrics-addr` to `--metrics-bind-address`. For more context see: https://book.kubebuilder.io/migration/v2vsv3.html#tldr-of-the-new-gov3-plugin

### CODE REVIEW

File: operator/apis/cascading/v1/groupversion_info.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 iteratec GmbH.
+Copyright 2021.
```
Comment (fuhrmeistery) [inline operator/apis/cascading/v1/groupversion_info.go]:
```Suggestion
Copyright 2021 iteratec GmbH.
```

File: operator/apis/cascading/v1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // +build !ignore_autogenerated
 
 /*
-Copyright 2020 iteratec GmbH.
```
Comment (fuhrmeistery) [inline operator/apis/cascading/v1/zz_generated.deepcopy.go]:
```Suggestion
Copyright 2021 iteratec GmbH.
```

File: operator/apis/execution/v1/groupversion_info.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 iteratec GmbH.
+Copyright 2021.
```
Comment (fuhrmeistery) [inline operator/apis/execution/v1/groupversion_info.go]:
```Suggestion
Copyright 2021 iteratec GmbH.
```

File: operator/apis/execution/v1/zz_generated.deepcopy.go [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // +build !ignore_autogenerated
 
 /*
-Copyright 2020 iteratec GmbH.
+Copyright 2021.
```
Comment (fuhrmeistery) [inline operator/apis/execution/v1/zz_generated.deepcopy.go]:
```Suggestion
Copyright 2021 iteratec GmbH.
```

File: operator/config/crd/kustomization.yaml [hunk 5]
```diff
@@ -2,37 +2,35 @@
...
-  - bases/execution.securecodebox.io_scheduledscans.yaml
-  - bases/cascading.securecodebox.io_cascadingrules.yaml
-# +kubebuilder:scaffold:crdkustomizeresource
+- bases/execution.securecodebox.io_scans.yaml
+- bases/execution.securecodebox.io_scantypes.yaml
+- bases/execution.securecodebox.io_parsedefinitions.yaml
+- bases/execution.securecodebox.io_scancompletionhooks.yaml
+- bases/cascading.securecodebox.io_cascadingrules.yaml
+- bases/execution.securecodebox.io_scheduledscans.yaml
+#+kubebuilder:scaffold:crdkustomizeresource
```
Comment (fuhrmeistery) [inline operator/config/crd/kustomization.yaml:15]: Should there not be an indent?

File: operator/config/crd/kustomization.yaml [hunk 6]
```diff
@@ -2,37 +2,35 @@
...
+#- patches/cainjection_in_scancompletionhooks.yaml
 #- patches/cainjection_in_cascadingrules.yaml
-# +kubebuilder:scaffold:crdkustomizecainjectionpatch
+#- patches/cainjection_in_scheduledscans.yaml
+#+kubebuilder:scaffold:crdkustomizecainjectionpatch
 
 # the following config is for teaching kustomize how to do kustomization for CRDs.
 configurations:
-  - kustomizeconfig.yaml
+- kustomizeconfig.yaml
```
Comment (fuhrmeistery) [inline operator/config/crd/kustomization.yaml:40]: Should there not be an indent?

File: operator/config/default/kustomization.yaml [hunk 7]
```diff
@@ -16,21 +16,25 @@ bases:
...
 #- ../prometheus
 
 patchesStrategicMerge:
-  # Protect the /metrics endpoint by putting it behind auth.
-  # If you want your controller-manager to expose the /metrics
-  # endpoint w/o any authn/z, please comment the following line.
+# Protect the /metrics endpoint by putting it behind auth.
+# If you want your controller-manager to expose the /metrics
+# endpoint w/o any authn/z, please comment the following line.
 - manager_auth_proxy_patch.yaml
```
Comment (fuhrmeistery) [inline operator/config/default/kustomization.yaml:35]: Should there not be an indent?

File: operator/hack/boilerplate.go.txt [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 iteratec GmbH.
+Copyright 2021.
```
Comment (fuhrmeistery) [inline operator/hack/boilerplate.go.txt]:
```Suggestion
Copyright 2021 iteratec GmbH.
```

File: operator/main.go [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 iteratec GmbH.
+Copyright 2021.
```
Comment (fuhrmeistery) [inline operator/main.go]:
```Suggestion
Copyright 2021 iteratec GmbH.
```

File: operator/config/rbac/role.yaml [hunk 10]
```diff
@@ -1,6 +1,3 @@
-# SPDX-FileCopyrightText: 2020 iteratec GmbH
-#
-# SPDX-License-Identifier: Apache-2.0
```
Comment (fuhrmeistery) [inline operator/config/rbac/role.yaml:3]: Was this intentional?

File: operator/config/crd/kustomization.yaml (hunk 5 above)
Comment (J12934) [reply operator/config/crd/kustomization.yaml:15]: Doesn't need to be.
Would prefer not to change it as this file was auto generated and might be unindented again on the next generation run.

File: operator/config/rbac/role.yaml (hunk 10 above)
Comment (J12934) [reply operator/config/rbac/role.yaml]: No 😄

File: operator/Makefile [hunk 11]
```diff
@@ -1,8 +1,8 @@
 
 # Image URL to use all building/pushing image targets
-IMG ?= securecodebox/operator:latest
+IMG ?= controller:latest
```
[the remark this answers is not in the document]
Comment (J12934) [reply operator/Makefile:6]: Changed back 👍

**Review by fuhrmeistery — APPROVED** [review]

**Comment by J12934** [comment]
🥳

**Pull request merged**